crates/bevy_asset/src/asset_server.rs

```diff
@@ -374,7 +391,12 @@ impl AssetServer {
         }
 
         let mut handles = Vec::new();
-        for child_path in self.server.asset_io.read_directory(path.as_ref())? {
+        for child_path in self
+            .server
+            .asset_io
+            .read_directory(path.as_ref())?
+            .filter(|p| predicate(p.as_path()))
```

Comment:
Optionally we could change this to take 2 functions.
One that filters directories, and one that filters files?